.github/workflows/preflight_env.sh

```diff
@@ -8,7 +8,7 @@ set -e
 #   - XCODE_VERSION: The version of Xcode to use.
 
 # GitHub runners are hitting 'module not found pkg_resources' required by prepare_sim.py
-pip3 install setuptools --break-system-packages
+pip3 install setuptools==69.5.1 --break-system-packages
```

Comment:
Unrelated to the other changes in this PR. Fixes this error happening on CI ([see repro](https://github.com/bazel-ios/rules_ios/actions/runs/9387193179/job/25849545106?pr=874)):
```sh
/Users/runner/work/rules_ios/rules_ios/tools/tests/prepare_sim.py:9: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import packaging
Traceback (most recent call last):
  File "/Users/runner/work/rules_ios/rules_ios/tools/tests/prepare_sim.py", line 9, in <module>
    from pkg_resources import packaging
ImportError: cannot import name 'packaging' from 'pkg_resources' (/opt/homebrew/lib/python3.12/site-packages/pkg_resources/__init__.py)
```

(found a thread online where someone was suggesting stick to this version to fix it)